std/format.d

```diff
@@ -4423,6 +4423,9 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
     static if (__traits(hasMember, T, "toString") && isSomeFunction!(val.toString))
         static assert(!__traits(isDisabled, T.toString), T.stringof ~
             " cannot be formatted because its `toString` is marked with `@disable`");
+    static if (__traits(hasMember, T, "toString") && !hasToString!(T, Char))
+        pragma(msg, T.stringof ~
+            " cannot be formatted because its `toString` cannot be instantiated.");
```

Comment:
Is there any possibility of getting a more informative error message here -- i.e. one that explains _why_ `toString` cannot be instantiated?  It would be sufficient to pass on the error message that would be produced if the problematic code were being compiled directly.

Reply:
Afraid not - there's [so many different ways to call `toString`](https://github.com/dlang/phobos/blob/master/std/format.d#L4024), and very hard (possibly no way) to tell which overloads are implemented and failed to instantiate. This also means we can't call `toString` to get the correct error messages.